src/indexer/segment_writer.rs

```diff
@@ -308,29 +317,62 @@ fn write(
     fast_field_writers: &FastFieldsWriter,
     fieldnorms_writer: &FieldNormsWriter,
     mut serializer: SegmentSerializer,
+    docid_map: Option<&DocidMapping>,
 ) -> crate::Result<()> {
     if let Some(fieldnorms_serializer) = serializer.extract_fieldnorms_serializer() {
-        fieldnorms_writer.serialize(fieldnorms_serializer)?;
+        fieldnorms_writer.serialize(fieldnorms_serializer, docid_map)?;
     }
     let fieldnorm_data = serializer
         .segment()
         .open_read(SegmentComponent::FieldNorms)?;
     let fieldnorm_readers = FieldNormReaders::open(fieldnorm_data)?;
-    let term_ord_map =
-        multifield_postings.serialize(serializer.get_postings_serializer(), fieldnorm_readers)?;
-    fast_field_writers.serialize(serializer.get_fast_field_serializer(), &term_ord_map)?;
+    let term_ord_map = multifield_postings.serialize(
+        serializer.get_postings_serializer(),
+        fieldnorm_readers,
+        docid_map,
+    )?;
+    fast_field_writers.serialize(
+        serializer.get_fast_field_serializer(),
+        &term_ord_map,
+        docid_map,
+    )?;
+    // finalize temp docstore and create version, which reflects the docid_map
+    if let Some(docid_map) = docid_map {
+        let store_write = serializer
+            .segment_mut()
+            .open_write(SegmentComponent::Store)?;
+        let mut store_writer = StoreWriter::new(store_write);
+        std::mem::swap(&mut serializer.store_writer, &mut store_writer);
```

Comment:
This is more a `mem::replace` than a `mem::swap` job here. I think clippy sends a warning.

Reply:
clippy doesn't complain, but yes this is a better version